### DISCUSSION TITLE
Scroll area on the drawer

### DISCUSSION THREAD

**Discussion by AmosSzust** [quasarframework/quasar#16835] [General - Components / Directives / etc]
I have a drawer with some data on it and I want the rest of the height to contain a timeline with a scrollbar, meaning the scrollbar will be just for the timeline component. I wanted to use QScrollArea but it must have px for height in order to work. Is there an easy way to make the scroll area work dynamically or maybe calculate the height somehow?

Thanks.

**Comment by AmosSzust** [comment]
Any kind of help or a pointer to where to look will be appreciated, thanks

**Comment by pratik227** [comment]
Use 'fit' with scroll area.

```
<q-scroll-area class="fit">
  <div class="q-pa-sm">
    <div v-for="n in 50" :key="n">Drawer {{ n }} / 50</div>
  </div>
</q-scroll-area>
```

Example - https://quasar.dev/layout/drawer/#basic

**Reply by AmosSzust** [reply]
It doesn't work if there are other components above the qscrollarea component inside the drawer. It just shows nothing, I have to give it a fixed px size for it to show something.

Also notice that the demo is using <q-layout view="hHh Lpr lff" **container style="height: 300px"** />

**Reply by pdanpdan** [reply]
make a codepen

**Reply by AmosSzust** [reply]
Here it is, I set a fixed height to see the content, I need the height of the scrollarea to take the remaining height.
https://codepen.io/amosss/pen/LYaXxVO?editors=101

**Reply by pdanpdan** [reply]
https://codepen.io/pdanpdan/pen/zYbMNpd?editors=1100

**Reply by AmosSzust** [reply]
Thanks, I guess I'm missing something because I tried following what you did but still qScrollArea's height is 0,
![image](https://github.com/quasarframework/quasar/assets/100950035/b8a7c23c-2cd6-4716-ac5d-d6c73ab44bec)

In my code, everything I wrote is inside a qDrawer and I wonder if something that is defined on the drawer level's classes, makes the difference. This is the code in the layout file.
```
    <q-drawer
      v-model="drawerLeft"
      show-if-above
      elevated
      class="bg-light-blue-10 text-white"
    >
      <drawer-content />
    </q-drawer>

```
